features/support/capybara_initializer.rb

```diff
@@ -0,0 +1,68 @@
+class CapybaraInitializer
+
+  attr_accessor :headless, :context
+  alias :headless? :headless
+
+  def initialize
+    @headless = true
+  end
+
+  def call
+    # Capybara defaults to XPath selectors rather than Webrat's default of CSS3. In
+    # order to ease the transition to Capybara we set the default here. If you'd
+    # prefer to use XPath just remove this line and adjust any selectors in your
+    # steps to use the XPath syntax.
+    Capybara.default_selector = :css
+
+    # Increase default wait time for asynchronous JavaScript requests from 2 to 5s
+    # see section on Asynchronous JavaScript here: https://github.com/jnicklas/capybara
+    Capybara.default_max_wait_time = 5
+
+    # Add some Capybara config if we are running
+    # Chrome on the docker host machine.
+    if !headless? && docker?
+      Capybara.server_host = '0.0.0.0'
+      Capybara.server_port = '43447'
```

Comment:
Ah, so this is where a random external port would be set.  So if the container could discover its external port that would work here.

Reply:
Yes, that is correct. However, see my response above :point_up: